File: Data Structures and Algorithms/LeetCode/202. Happy Number.cs
public class Solution {
    public bool IsHappy(int n) {
        int count=0;
        string ns = "";
        int newn=0;
        
        while(n != 1){
            count++;
            ns=n.ToString();
           foreach(char i in ns){
                newn= newn+ (int)Char.GetNumericValue(i)*(int)Char.GetNumericValue(i);
            }
            n=newn;
            newn=0;
            
            
            if(count>1000) return false;
        }
        return true;  
    }
}

File: Data Structures and Algorithms/LeetCode/202. Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        num=0
        i=0
        setA=set()
        while True:
            for char in str(n):
                num+=int(char)**2
            n=str(num)
            
            if num in setA:
                return False
            setA.add(num)

            if num==1: 
                return True
            num=0